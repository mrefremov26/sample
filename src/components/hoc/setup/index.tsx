import React from 'react';

import { withAuthChecker } from '@hoc/withAuthChecker';
import { withModals } from '@hoc/withModals';
import { withProviders } from '@hoc/withProviders';
import { withScrollReset } from '@hoc/withScrollReset';

export const setup = (Component: React.FC<any>): any => {
  return compose(Component, React.memo, withScrollReset, withAuthChecker, withModals, withProviders);
};

function compose(Component: React.FC, ...functions: Array<(...args: any[]) => React.FC>) {
  return functions.reduce((acc: React.FC, f: (...args: any[]) => React.FC) => f(acc), Component);
}
