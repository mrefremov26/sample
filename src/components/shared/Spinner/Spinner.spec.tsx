import * as React from 'react';
import renderer from 'react-test-renderer';

import { Spinner } from '.';
import { setup } from '@hoc/setup';

describe('Spinner renders correctly', () => {
  it('snapshot', () => {
    const Component = setup(Spinner);
    const tree = renderer.create(<Component />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
