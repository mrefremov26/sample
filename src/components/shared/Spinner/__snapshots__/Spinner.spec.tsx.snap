// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner renders correctly snapshot 1`] = `
<svg
  fill="none"
  height="18"
  viewBox="0 0 18 18"
  width="18"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M17 9C17 7.07407 16.3052 5.21278 15.0432 3.75793C13.7812 2.30307 12.0368 1.35231 10.1302 1.08023C8.22355 0.808155 6.28277 1.23303 4.66423 2.27683C3.04568 3.32063 1.85801 4.91329 1.31932 6.76234C0.78062 8.6114 0.927057 10.5927 1.73174 12.3425C2.53642 14.0923 3.94532 15.493 5.69973 16.2875C7.45415 17.0821 9.43629 17.217 11.2822 16.6676"
    stroke="#505F79"
    strokeLinecap="round"
    strokeWidth="2"
  />
</svg>
`;
